bootstrap/bootstrap.go

```diff
@@ -173,8 +195,111 @@ func (b *Bootstrap) Cancel() error {
 	return nil
 }
 
+// extractTraceCtx pulls encoded distributed tracing information from the env vars.
+// Note: This should match the injectTraceCtx code in shell.
+func (b *Bootstrap) extractTraceCtx() opentracing.SpanContext {
```

Comment:
This and shell.injectTraceCtx isn't really useful right now. I can remove it and save the code for a future change. This was a first stab at getting distributed tracing, but it doesn't work because env vars aren't propagated to dynamically triggered pipelines/jobs. We'd have to modify some other code. But it'd be pretty sweet to see our main pipeline job trace show links to all the dynamic pipelines we spin up. This can be a v2 thing IMO.

Reply:
Yeah, being able to see traces for all downsteam builds would be _the dream_! Sounds like a good idea to hold off though and keep the scope small for this first PR.

Reply:
Got it working! Pushed to this branch.
![Screenshot from 2020-08-20 16-23-13](https://user-images.githubusercontent.com/29210237/90835511-ab803500-e301-11ea-8e9e-c98e9f5de4b3.png)